### PR TITLE
[stable/nginx-ingress] Allow to configure source IP preservation on Kubernetes 1.7

### PR DESCRIPTION
**Pull request by mstrzele**
Unfortunately, after upgrading Kubernetes to 1.7, I'm no longer able to set `service.beta.kubernetes.io/external-traffic`.

According to [documentation](https://kubernetes.io/docs/tasks/access-application-cluster/create-external-load-balancer/#feature-availability), the annotation is deprecated and the field under `service.spec` should be used.

This pull request adds two values to controller's service: `externalTrafficPolicy` and `healthCheckNodePort`

### CODE REVIEW

**Comment by mstrzele** [comment]
Probably duplicate of #1542. The only difference is the `healthCheckNodePort` value.

File: stable/nginx-ingress/values.yaml [hunk 1]
```diff
@@ -91,6 +91,9 @@ controller:
       http: ""
       https: ""
 
+    # externalTrafficPolicy: Local
```
Comment (mgoodness) [inline stable/nginx-ingress/values.yaml]: Un-comment this and set to `""`

File: stable/nginx-ingress/values.yaml [hunk 2]
```diff
@@ -91,6 +91,9 @@ controller:
...
+    # healthCheckNodePort: 32080
```
Comment (mgoodness) [inline stable/nginx-ingress/values.yaml]: Un-comment and set to `0` (zero)

**Review by mgoodness — CHANGES_REQUESTED** [review]
Just a couple minor changes. Be sure to rebase & re-bump chart version.

File: stable/nginx-ingress/templates/controller-service.yaml [hunk 3]
```diff
@@ -57,3 +57,9 @@ spec:
     component: "{{ .Values.controller.name }}"
     release: {{ .Release.Name }}
   type: "{{ .Values.controller.service.type }}"
+{{- if .Values.controller.service.externalTrafficPolicy }}
```
Comment (mgoodness) [inline stable/nginx-ingress/templates/controller-service.yaml]: Make this `{{- if and (ge .Capabilities.KubeVersion.Minor "7") (.Values.controller.service.externalTrafficPolicy) }}` so v1.6 (and lower) is still supported.

File: stable/nginx-ingress/templates/controller-service.yaml [hunk 4]
```diff
@@ -57,3 +57,9 @@ spec:
...
+  externalTrafficPolicy: "{{ .Values.controller.service.externalTrafficPolicy }}"
+{{- end }}
+{{- if .Values.controller.service.healthCheckNodePort }}
```
Comment (mgoodness) [inline stable/nginx-ingress/templates/controller-service.yaml]: Make this `{{- if and (ge .Capabilities.KubeVersion.Minor "7") (.Values.controller.service. healthCheckNodePort) }}` so v1.6 (and lower) is still supported.

**Review by mgoodness — DISMISSED** [review]
Couple more changes. Sorry.

**Pull request merged**